### PR TITLE
support optional mysql usage

### PR DESCRIPTION
**Pull request by travisghansen**
Leaves mysql starting by default.  Only ignores if env var ```START_MYSQL=false```.

### CODE REVIEW

**Comment by jasonblais** [comment]
Thanks @travisghansen for the pull request!

 Per the [**CONTRIBUTING.md**](https://github.com/mattermost/platform/blob/master/CONTRIBUTING.md#submitting-a-pull-request) file displayed when you created this pull request, we need to add you to the list of [approved contributors](https://docs.google.com/spreadsheets/d/1NTCeG-iL_VS9bFqtmHSfwETo5f-8MQ7oMDE5IUYJi_Y/pubhtml?gid=0&single=true) for the Mattermost project.

**Please help complete the Mattermost [contribution license agreement](http://www.mattermost.org/mattermost-contributor-agreement/)?**

This is a standard procedure for many open source projects. Your form should be processed within 24 hours and reviewers for your pull request will be able to proceed.

Please let us know if you have any questions.

We are very happy to have you join our growing community! If you're not yet a member, please consider joining our [Contributors community channel](https://pre-release.mattermost.com/core/channels/tickets) to meet other contributors and discuss new opportunities with the core team.

**Comment by travisghansen** [comment]
Can this get looked at again?

File: docker-entry.sh [hunk 1]
```diff
@@ -2,13 +2,15 @@
 # Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
 # See License.txt for license information.
 
-echo "Starting MySQL"
-/entrypoint.sh mysqld &
+if [ "x${START_MYSQL,,}" != "xfalse" ]; then
```
Comment (cpanato) [inline docker-entry.sh]: Sorry for the long delay here.
this is still relevant?
I dont see a use case for that. can you explain why we need that? thanks again

**Comment by travisghansen** [comment]
@cpanato because running multiple services in a single container is considered 'bad'.  If you run an external MySQL instance (container or not) this process is just unnecessary bloat.

**Comment by cpanato** [comment]
@travisghansen yes, agreed. However, the idea for this image here is not to use in production, is just to test the app in an easy way.
For production deployment, you will need to use the https://github.com/mattermost/mattermost-docker and follow the guides in docs.mattermost.com

**Comment by travisghansen** [comment]
Fair enough.  Not sure that existed at the time this issue was created :)

**Comment by travisghansen** [comment]
@cpanato go ahead and close.

**Pull request closed without merging**